libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -433,6 +433,25 @@ void ContractLevelChecker::checkHashCollisions(ContractDefinition const& _contra
 			);
 		hashes.insert(hash);
 	}
+
+	map<uint32_t, SourceLocation> errorHashes;
+	// TODO all used errors?
```

Comment:
All defined errors seems good. You can still use them when inheriting, can't you? And then only erroring out late would be weird.